src/inmanta/server/services/codeservice.py

```diff
@@ -125,7 +104,6 @@ async def get_code(self, env: data.Environment, code_id: int, resource: str) ->
         sources = {}
         if code.source_refs is not None:
             for code_hash, (file_name, module, req) in code.source_refs.items():
-                content = self.file_slice.get_file_internal(code_hash)
-                sources[code_hash] = (file_name, module, content.decode(), req)
+                sources[code_hash] = (file_name, module, "", req)
```

Comment:
I think this could use a comment. We just don't use the column anymore but we keep it for backwards compatibility? Perhaps we should also add this to the method docstring?

Reply:
I keep it here to make sure that we do not accidentally break code or tests. The code that actually uses this, the agent code loading, has been updated.

Reply:
but it is in fact a breaking change that will break older agents silently? 

I wonder if it would not be better to make it very obvious that this is broken? 

or even make a new API endpoint, and leave this one for backward compatibility and get rid of it in iso6?

Reply:
@bartv I changed this to raise a 400 when the source is not valid utf8

Reply:
that is in the case it is byte compiled?

Reply:
or any other binary file, yes 

I added it to the testcase